peer/peer_storage_test.go

```diff
@@ -0,0 +1,75 @@
+package peer
+
+import (
+	"os"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/stretchr/testify/require"
+)
+
+// TestPeerStorageDB tests the PeerStorageDB functionality.
+func TestPeerStorageDB(t *testing.T) {
+	rt := require.New(t)
+
+	file, err := os.CreateTemp("", "*.db")
+	rt.NoError(err)
+	t.Cleanup(func() {
+		rt.NoError(file.Close())
+		rt.NoError(os.Remove(file.Name()))
+	})
+
+	dbPath := file.Name()
+	db, err := kvdb.Open(
+		kvdb.BoltBackendName, dbPath, true, kvdb.DefaultDBTimeout,
+	)
```

Comment:
This should be wrapped with something that also initializes it and makes it ready for use as mentioned elsewhere.

Reply:
Do you mean that I should wrap the db in a struct? Personally  I do not see the need

Reply:
Ignore the comment here, address the one above. The point is that the function `InitPeerStorageDB` is a bad design pattern and should be removed. We should not be returning handles to uninitialized resources. There's not good reason for it. Initialize on creation, if it's already initialized, swallow the error and move on with life.